### PR TITLE
docs: improve `x/upgrade` docs

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3223

x/upgrade README format inspired by https://docs.cosmos.network/v0.50/build/spec/SPEC_MODULE

## FLUPs
1. Can we move some of the unit tests from tally_test.go to keeper_test.go?
2. Why does ResetTally only delete versions less than the current version? Why not delete all versions in the map?

### CODE REVIEW

File: x/upgrade/README.md [hunk 1]
```diff
@@ -1,23 +1,77 @@
...
+
+N/A
+
+## Future Improvements
+
+N/A
+
+## Tests
+
+N/A
```
Comment (cmwaters) [inline x/upgrade/README.md:70]: Is it necessary to keep the sections that are N/A?

Comment (rootulp) [reply x/upgrade/README.md:70]: Not strictly necessary in my opinion. Can definitely remove

Comment (rootulp) [reply x/upgrade/README.md]: https://github.com/celestiaorg/celestia-app/pull/3224/commits/cbea0ebb070b29c9c04914fc039ac961886fd4e9

**Review by cmwaters — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Comment by cmwaters** [comment]
> Can we move some of the unit tests from tally_test.go to keeper_test.go?

Yeah we can if you want

> Why does ResetTally only delete versions less than the current version? Why not delete all versions in the map?

I guess the one case is if you've already voted for v3 before upgrading to v2. Then you'd have to revote because it would be deleted. But this is not really an expected use case so I have nothing against deleting all the versions in the map

**Pull request merged**